argilla/docs/how_to_guides/export.md

```diff
@@ -0,0 +1,160 @@
+
+# Importing and exporting datasets and records
+
+This guide provides an overview of how to import and export your dataset or its records to Python, your local disk, or the Hugging Face Hub.
+
+In Argilla, you can import/export two main components of a dataset:
+- The dataset's complete configuration defined in `rg.Settings`. This is useful if your want to share your feedback task or restore it later in Argilla.
+- The records stored in the dataset, including `Metadata`, `Vectors`, `Suggestions`, and `Responses`. This is useful if you want to use your dataset's records outside of Argilla.
+
+Check the [Dataset - Python Reference](../reference/argilla/datasets/dataset.md) to see the attributes, arguments, and methods of the export `Dataset` class in detail.
+
+## Export an `rg.Dataset` from Argilla
+
+First, we will go through exporting a complete dataset from Argilla. This includes the dataset's setting and records. All of these methods use the `rg.Dataset.from_*` and `rg.Dataset.to_*` methods.
+
+### Push an Argilla dataset to the Hugging Face Hub
+
+You can push a dataset from Argilla to the Hugging Face Hub. This is useful if you want to share your dataset with the community or version control it. You can push the dataset to the Hugging Face Hub using the `rg.Dataset.to_hub` method.
+
+```python
+import argilla as rg
+
+client = rg.Argilla(api_url="<api_url>", api_key="<api_key>")
+dataset = client.datasets(name="my_dataset")
+dataset.to_hub(repo_id="<repo_id>")
+```
+
+!!! note "With or without records"
+    The example above will push the dataset's `Settings` and records to the hub. If you only want to push the dataset's configuration, you can set the `with_records` parameter to `False`. This is useful if you're just interested in a specific dataset template or you want to make changes in the dataset settings and/or records.
+
+    ```python
+    dataset.to_hub(repo_id="<repo_id>", with_records=False)
+    ```
+    With the dataset's configuration you could then make changes to the dataset's settings, or add records via the `datasets` package.
+
+    ```python
+    hf_dataset = load_dataset("<repo_id>")
+    dataset.log(hf_dataset)
+    ```
+
+### Pull an Argilla dataset from the Hugging Face Hub
+
+You can pull a dataset from the Hugging Face Hub to Argilla. This is useful if you want to restore a dataset and its configuration. You can pull the dataset from the Hugging Face Hub using the `rg.Dataset.from_hub` method.
+
+```python
+
+import argilla as rg
+
+client = rg.Argilla(api_url="<api_url>", api_key="<api_key>")
+dataset = rg.Dataset.from_hub(repo_id="<repo_id>")
+```
+
+Note that this approach loads the configuration from the repo and stores the records. If you only want to load records, use the `load_dataset` method of the `datasets` package, and pass the dataset to `rg.Dataset.log` method. This enables you to configure your own dataset and reuse existing Hub datasets. See the [guide on records](record.md) for more information.
```

Comment:
I tried to rewrite this part but I'm not super happy with the result for the second part (if you want to load records...), maybe a tip with an explicit code snippet would help but I also feel we're kind of repeating ourselves with the load_dataset --> log approach. So maybe a shorter sentence and the pointer to guide on records

Reply:
I've rewritten and restructured [this patch of prose. ](https://github.com/argilla-io/argilla/blob/4f4b1144a3f0f69c2942e3a3842a4f97bbf59268/argilla/docs/how_to_guides/import_export.md?plain=1#L51). I tried to just make it more explicit because, as you say, the record guide covers this.